src/api/local-storage-provider.model.ts

```diff
@@ -2,67 +2,136 @@
+    });
+    if (!_.isEmpty(prevMonthShifts)) {
+      await this.updateMonthRevision(
+        type,
+        getScheduleKey(scheduleDataModel).prevMonthKey,
+        prevMonthShifts,
+        "TAIL"
+      );
+    }
+
+    const nextMonthShifts = {};
+    Object.keys(scheduleDataModel.shifts).forEach((key) => {
+      nextMonthShifts[key] = scheduleDataModel.shifts[key].slice(lastMonthDayIdx);
+    });
+    if (!_.isEmpty(nextMonthShifts)) {
+      await this.updateMonthRevision(
+        type,
+        getScheduleKey(scheduleDataModel).nextMonthKey,
+        nextMonthShifts,
+        "HEAD"
+      );
+    }
+  }
+
+  async updateMonthRevision(
     type: RevisionType,
-    schedule: ScheduleDataModel
-  ): ThunkFunction<ScheduleDataModel> {
-    return async (dispatch): Promise<void> => {
-      if (isScheduleModelEmpty(schedule)) {
-        return;
-      }
-      const id = this.getScheduleId(schedule);
-      let revision = "";
-      try {
-        const document = await this.storage.get(id);
-        revision = document._rev;
-      } catch {}
+    monthKey: ScheduleKey,
+    updateShifts: ShiftInfoModel,
+    updatePosition: UpdatePosition
+  ): Promise<void> {
+    try {
+      const document = await this.storage.get(monthKey.dbKey);
+      const revision = document._rev;
+
+      const newShifts = _.cloneDeep(document.data.shifts);
+
+      Object.keys(document.data.shifts).forEach((key) => {
+        if (updatePosition === "TAIL") {
+          newShifts[key].splice(0, updateShifts[key].length, ...updateShifts[key]);
+        } else {
+          newShifts[key].splice(
+            newShifts[key].length - updateShifts[key].length,
+            newShifts[key].length,
+            ...updateShifts[key]
+          );
+        }
+      });
+      document.data.shifts = newShifts;
       this.storage.put({
         _rev: revision,
-        _id: id,
-        data: schedule,
+        _id: monthKey.dbKey,
+        data: document.data,
         revisionType: type,
       });
-      const action = ScheduleDataActionCreator.setSchedule(schedule);
-      dispatch(action);
-    };
+    } catch {
+      //TODO: Something should be done here :)
+    }
   }
 
-  getScheduleRevision(filter: RevisionFilter): ThunkFunction<ScheduleDataModel> {
-    return async (dispatch): Promise<void> => {
-      const revisions = await this.storage.allDocs({ include_docs: true });
-      const result = revisions.rows.find((r) => {
-        const { year, month_number } = r.doc?.data.schedule_info ?? {};
-        return (
-          month_number === filter.validityPeriod.month &&
-          year === filter.validityPeriod.year &&
-          r.doc?.revisionType === filter.revisionType
-        );
-      });
-      if (!result?.doc) {
-        return;
+  async getMonthRevision(filter: RevisionFilter): Promise<MonthDataModel | undefined> {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    const revisions = await this.storage.allDocs({ include_docs: true });
```

Comment:
w przypadku takich rzeczy, że backend używa snake_case a front camelCase są gotowe libki to translacji

Reply:
kurde to by się nam bardzo przydało, bo cały czas się to pojawia w projekcie i jest strasznie denerwujące. Odwlekaliśmy wdrażanie tego, żeby nie robić Wielkiego Refacotru, ale po zamknięciu milestona to pierwsza rzecz, którą wprowadzimy. Jak masz coś sprawdzonego to zapodaj :)

Reply:
No ja zrobiłem w axiosie takie ustawienia i to samo mi translatuje w tą i spowrotem
```
return axios.create({
  baseURL: API_URL,
  transformResponse: [...axios.defaults.transformResponse, (data) => humps.camelizeKeys(data)],
  transformRequest: [
    (data) => (data instanceof FormData ? data : humps.decamelizeKeys(data)),
    ...axios.defaults.transformRequest,
  ],
});
```

Reply:
Luks rozwiązanie, to już nie w tym tasku, ale może faktycznie przed Wilekim Refactorinigiem XD @Qwebeck co myślisz?

Reply:
Tutaj jest ta zaleta, że wszystko co było już zamienione, zostanie bez zmian, także jak wszystko było wysyłane do backendu u snake to po dodaniu nic się nie zmieni